tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+    create_aof_dir $aof_dirpath
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof seq 1 type b\n"
+    }
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can't load data when the manifest contains the old AOF file name but the file does not exist in server dir and aof dir" {
+            wait_for_condition 100 50 {
+                ! [is_alive $srv]
+            } else {
+                fail "AOF loading didn't fail"
+            }
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+
+    # Tests20: Multi Part AOF can upgrade when when two redis share the same server dir
+    create_aof $aof_old_version_filepath {
+        append_to_aof [formatCommand set k1 v1]
+        append_to_aof [formatCommand set k2 v2]
+        append_to_aof [formatCommand set k3 v3]
+    }
+
+    create_aof $aof_old_version_filepath2 {
+        append_to_aof [formatCommand set k4 v4]
+        append_to_aof [formatCommand set k5 v5]
+        append_to_aof [formatCommand set k6 v6]
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can upgrade when when two redis share the same server dir 1" {
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
+
+            assert_equal v1 [$client get k1]
+            assert_equal v2 [$client get k2]
+            assert_equal v3 [$client get k3]
+
+            assert_equal 0 [$client exists k4]
+            assert_equal 0 [$client exists k5]
+            assert_equal 0 [$client exists k6]
+            
+            assert_aof_manifest_content $aof_manifest_filepath  {
+                {file appendonly.aof seq 1 type b} 
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+            
+            $client bgrewriteaof
+            while 1 {
+                if {[status $client aof_rewrite_in_progress] eq 1} {
+                    if {$::verbose} {
+                        puts -nonewline "\nWaiting for background AOF rewrite to finish... "
+                        flush stdout
+                    }
+                    after 1000
+                } else {
+                    break
+                }
+            }
+
+            assert_equal OK [$client set k v]
+
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.2.base.rdb seq 2 type b} 
+                {file appendonly.aof.2.incr.aof seq 2 type i}
+            }
+
+            set d1 [$client debug digest]
+            $client debug loadaof
+            set d2 [$client debug digest]
+            assert {$d1 eq $d2}
+        }
+    }
+
+    start_server [list overrides [list dir $server_path appendonly yes appendfilename appendonly.aof2 appenddirname appendonlydir]] {
+        test "Multi Part AOF can upgrade when when two redis share the same server dir 2" {
```

Comment:
i think it would be better if the two servers are nested (running at the same time).
```tcl
test {name} {
    setup code....
    start_server 1 {
        start_server 2 {
            all the checks referring to [r -1 command] and [r 0 command]
        }
    }
}
```

Reply:
OK